sdks/python/apache_beam/typehints/trivial_inference.py

```diff
@@ -65,18 +65,27 @@ def instance_to_type(o):
   elif t == tuple:
     return typehints.Tuple[[instance_to_type(item) for item in o]]
   elif t == list:
-    return typehints.List[
-        typehints.Union[[instance_to_type(item) for item in o]]
-    ]
+    if len(o):
```

Comment:
nit: `len(o) > 0` is clearer IMO (easier to read quickly). Here and below.

Reply:
done